### PR TITLE
Add ability to get/set forced bed location. Resolves #955

### LINKED ISSUE #955: Add a way to force non-bed spawns with RespawnLocationData
Hello all, I've been working on a Graveyards plugin and have noticed that currently with [RespawnLocationData](https://jd.spongepowered.org/org/spongepowered/api/data/manipulator/mutable/entity/RespawnLocationData.html), there is no way to force a non-bed spawn location.  Whenever I offer a non-bed location it is rejected and the player is told that their home bed was missing or obstructed and they are respawned at the default spawn point in the world.  I've been told that forge had a force spawn position flag, is there anyway would could implement a similar feature in sponge?

Thanks!

### PR DESCRIPTION
**Pull request by simon816**
#961 on the bleeding edge

Changes `Vector3d` to `Tuple<Vector3d, Boolean>`
See #955

### CODE REVIEW

**Comment by Mumfrey** [comment]
I have a couple of problems with this, though they're easily addressed.

I'm generally against the use of `Tuple`s as a lazy way of doing what really needs to be a custom struct (just as @gabizou, I [made him change some the other day](https://github.com/SpongePowered/SpongeCommon/commit/67972897c9e4ba51c1493b2d40b15b134b130d4b)) as it suffers from the same type of issue as *[the boolean trap](http://ariya.ofilabs.com/2011/08/hall-of-api-shame-boolean-trap.html)* in that it instantly obfuscates the purpose of the data structure in question. This is not to say there aren't good uses of `Tuple`, but this isn't one of them.

Secondly, this also falls directly into the *boolean trap*, and it's precisely because the use of the `Tuple` hides the purpose of the `bool`.

What you're trying to indicate is that a particular `Vector` is "forced" and pairing it with a boolean using a `Tuple` is just smelly. You should either:

* Create a struct to hold your vec3 and bool, for example `SpawnLocation`
* Extend `Vector3` with a subclass which implements (for example) a new interface `IForceable` (or whatever, maybe `IEnforceable` might be better) containing an `isForced()` accessor and use the new subclass `ForcedLocation`

**Comment by simon816** [comment]
I have addressed @Mumfrey's issue by creating a `RespawnLocation` interface

File: src/main/java/org/spongepowered/api/entity/living/player/RespawnLocation.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+import org.spongepowered.api.world.World;
+
+import java.util.Optional;
+import java.util.UUID;
+
+/**
+ * Represents a location for a player to respawn in in a particular world.
+ * Instances can be obtained using {@link RespawnLocationData}.
+ */
+public interface RespawnLocation {
```
Comment (kashike) [inline src/main/java/org/spongepowered/api/entity/living/player/RespawnLocation.java]: Does this need to be represented as an interface, or can it be class?

Comment (simon816) [inline src/main/java/org/spongepowered/api/entity/living/player/RespawnLocation.java]: I suppose it could just be a class

**Pull request merged**